packages/block-library/src/template-part/edit/utils/transformers.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress dependencies
+ */
+import { createBlock, parse } from '@wordpress/blocks';
+
+/**
+ * Converts a widget entity record into a block.
+ *
+ * @param {Object} widget The widget entity record.
+ * @return {Object} a block (converted from the entity record).
+ */
+export function transformWidgetToBlock( widget ) {
```

Comment:
Let's re-use the `transformWidgetToBlock` function that's in `@wordpress/edit-widgets` by moving it to `@wordpress/widgets`. We should also add some unit tests—not sure why there aren't any already.

Reply:
It makes sense, but we'll have to make the original method more generic and remove `addWidgetIdToBlock`.

Reply:
Right, yeah. `addWidgetIdToBlock` should probably always be called by the caller of `transformWidgetToBlock`.

```js
// instead of edit-widgets doing this:
const block = transformWidgetToBlock( widget );
// should be this:
const block = addWidgetIdToBlock( transformWidgetToBlock( widget ) );
```

Reply:
I just really don't want us to duplicate this code yet again. We already have a near-copy in `@wordpress/customise-widgets` called `widgetToBlock` 😛

Reply:
Hm, probably not. Does this feature make sense outside of the site editor? Should it be a `Fill` added by `@wordpress/edit-site`?

Reply:
@mtias, should we limit this feature to the Site Editor?

Reply:
@Mamaduka sorry, missed the ping. It's a site editing specific flow, for sure. But have to keep in mind that you can get into editing the template of a page from the post editor, and we probably need the flow there.